external/common/lapack/FindTargetOpenMP.cmake

```diff
@@ -58,71 +51,93 @@ if (OpenMP_LIBRARIES AND OpenMP_FLAGS)
+                if (CMAKE_${_lang}_COMPILER_ID MATCHES GNU)
+                    set(OpenMP_${_lang}_FLAGS "-fopenmp")
+                    set(OpenMP_${_lang}_LIB_NAMES "gomp;pthread")
+
+                elseif (CMAKE_${_lang}_COMPILER_ID MATCHES Intel)
+                    set(OpenMP_${_lang}_FLAGS "-qopenmp")
+                    set(OpenMP_${_lang}_LIB_NAMES "iomp5;pthread")
+
+                elseif (CMAKE_${_lang}_COMPILER_ID STREQUAL Clang)
+                    if (OpenMP_${_lang}_FLAG MATCHES "-fopenmp=libiomp5")
+                        set(OpenMP_${_lang}_FLAGS "-fopenmp=libiomp5")
+                        set(OpenMP_${_lang}_LIB_NAMES "iomp5")
+                    elseif (OpenMP_${_lang}_FLAG MATCHES "-fopenmp=libgomp")
+                        set(OpenMP_${_lang}_FLAGS "-fopenmp=libgomp")
+                        set(OpenMP_${_lang}_LIB_NAMES "gomp")
+                    else()
+                        set(OpenMP_${_lang}_FLAGS "-fopenmp=libomp")
+                        set(OpenMP_${_lang}_LIB_NAMES "omp")
+                    endif()
                 endif()
 
             endif()
 
-            find_omp_libs("TargetOpenMP_${_lang}" ${OpenMP_${_lang}_LIB_NAMES})
-            if (TargetOpenMP_${_lang}_LIBRARIES)
+            find_omp_libs("${PN}_${_lang}" ${OpenMP_${_lang}_LIB_NAMES})
+            if (${PN}_${_lang}_LIBRARIES)
                 add_library(OpenMP::OpenMP_${_lang} INTERFACE IMPORTED)
-                set_property (TARGET OpenMP::OpenMP_${_lang} PROPERTY INTERFACE_COMPILE_OPTIONS "$<$<COMPILE_LANGUAGE:${_lang}>:${OpenMP_${_lang}_FLAGS}>")
-                set_property (TARGET OpenMP::OpenMP_${_lang} PROPERTY INTERFACE_LINK_LIBRARIES "${TargetOpenMP_${_lang}_LIBRARIES}")
+                target_compile_options(OpenMP::OpenMP_${_lang} INTERFACE $<$<COMPILE_LANGUAGE:${_lang}>:${OpenMP_${_lang}_FLAGS}>)
+                target_link_libraries(OpenMP::OpenMP_${_lang} INTERFACE ${${PN}_${_lang}_LIBRARIES})
                 if (NOT ${PN}_FIND_QUIETLY)
-                    message (STATUS "OpenMP_${_lang} target constructed.")
+                    message (STATUS "OpenMP::OpenMP_${_lang} target constructed.")
                 endif()
             endif()
         endif()
     endforeach()
 endif()
 
-unset(_omp_target_for_sought_langs)
-add_library (OpenMP::OpenMP INTERFACE IMPORTED)
-foreach(_lang IN ITEMS C CXX Fortran)
-    if((NOT ${PN}_FIND_COMPONENTS AND CMAKE_${_lang}_COMPILER_LOADED) OR _lang IN_LIST ${PN}_FIND_COMPONENTS)
-        if (TARGET OpenMP::OpenMP_${_lang})
-            set(${PN}_${_lang}_FOUND 1)
-            set_property(TARGET OpenMP::OpenMP APPEND PROPERTY INTERFACE_LINK_LIBRARIES "OpenMP::OpenMP_${_lang}")
-        endif()
-        list(APPEND _omp_target_for_sought_langs "${PN}_${_lang}_FOUND")
-        set (${PN}_MESSAGE "Found TargetOpenMP ${${PN}_FINDLIST}: ${_ill}")  # only last lang
+add_library(OpenMP::OpenMP INTERFACE IMPORTED)
+set(${PN}_FOUND 1)
+foreach(_lang ${${PN}_FIND_COMPONENTS})
```

Comment:
Here the script should mimick this: https://github.com/Kitware/CMake/blob/master/Modules/FindOpenMP.cmake#L507-L519 and `${PN}_FOUND` should be set by `find_package_handle_standard_args`.

Reply:
@loriab I see your point, I will return the functionality.